### DISCUSSION TITLE
im confused🤷‍♂️

### DISCUSSION THREAD

**Discussion by itsmohammadh** [Cyfrin/foundry-full-course-cu#795] [General]
Hello everyone. I am currently in the lesson 8. By looking at the code, I understand very well what each part and everything is for, and I am able to understand the code. But when I write the code, I get confused and I can't finish it. I would be grateful if anyone has this experience and give me a suggestion.

**Comment by 0xwtfwtf** [comment]
My suggestion is just to keep going with the videos...I had no coding background and I was in a similar place but now that I am on Lesson 12, I am able to code the stuff myself before we even do it with Patrick. In Lesson 12, Patrick tells you a list of all the functions you will be coding, so I just paused the video and coded them myself. It will sink in if you just keep going.
